src/Common/src/CoreLib/System/IO/Stream.cs

```diff
@@ -766,9 +766,7 @@ public virtual int ReadByte()
         {
             byte[] oneByteArray = new byte[1];
             int r = Read(oneByteArray, 0, 1);
-            if (r == 0)
-                return -1;
-            return oneByteArray[0];
+            return r != 0 ? oneByteArray[0] : -1;
```

Comment:
All changes to Stream.cs should be reverted, as it's not actually compiled in corefx.

Reply:
Done https://github.com/dotnet/corefx/pull/36838/commits/7db79108c5254cbebaea33ff766032b253b1a217